libvast/src/detail/mmapbuf.cpp

```diff
@@ -83,6 +85,8 @@ mmapbuf::~mmapbuf() {
     munmap(map_, size_);
   if (fd_ != -1)
     close(fd_);
+  if (is_file_owner_)
+    std::remove(filename_.str().c_str());
```

Comment:
Why do we need `c_str` here?`std::path` has a constructor that takes a string: https://en.cppreference.com/w/cpp/filesystem/path/path, or do I miss something?

Reply:
@Neverlord We haven't converted `vast::path` to `std::path` yet, perhaps that's the reason.

My comment here would be: we haven't used the new filesystem API so far in VAST, this is the first use. I'd rather use the old stuff from `vast/filesystem.hpp` (here: `vast::rm`) and then migrate all filesystem operations at once. Mixing and matching results only in confusion.